api/src/main/java/org/openmrs/module/muzimafingerPrint/api/impl/MuzimafingerPrintServiceImpl.java

```diff
@@ -85,7 +87,36 @@ public PatientFingerPrintModel identifyPatient(String fingerprint) throws IOExce
         }
         if(patient == null)
             return null;
-        return new PatientFingerPrintModel(patient.getUuid(),fingerprint, patient.getId(), patient.getGivenName(), patient.getFamilyName(), patient.getGender());
+        else {
+            mfingerprint = getFingerprintByPatientUUID(patient.getUuid());
+            if (mfingerprint != null) {
+                patfingerprint = mfingerprint.getFingerprint();
+            }
+            return new PatientFingerPrintModel(patient.getUuid(), patfingerprint, patient.getId(), patient.getGivenName(), patient.getFamilyName(), patient.getGender());
+        }
+    }
+
+    public List<PatientFingerPrintModel> findPatients(String searchInput) {
+        List<PatientFingerPrintModel> patients = new ArrayList<PatientFingerPrintModel>();
+        List<Patient> searchResults = Context.getPatientService().getPatients(searchInput);
+        MuzimaFingerprint mfingerprint = null;
+        String fingerprint = null;
+        if(searchResults.size() != 0){
+            for(Patient patientSearched : searchResults) {
+                mfingerprint = getFingerprintByPatientUUID(patientSearched.getUuid());
+                if (mfingerprint != null) {
+                    fingerprint = mfingerprint.getFingerprint();
+                }
+                patients.add(new PatientFingerPrintModel(patientSearched.getUuid(),
```

Comment:
I think a better structure for the patient finger print model will help here. Searching patient list, then searching fingerprint, then creating the structure doesn't looks very efficient I think. Maybe need some refactoring on the structure of the model itself?